### PR TITLE
Initial configuration for kubevirt openshift-ci onboarding

### PR DESCRIPTION
**Pull request by dhiller**
Contains required changes for onboarding [KubeVirt](https://github.com/kubevirt/kubevirt/) to openshift-ci.

Current status: what we are currently doing is the following:
* create an initial configuration for KubeVirt master
* spin up an OpenShift cluster on Google Cloud with libvirt enabled (to have KVM available and avoid emulation of virtualization)
* deploy KubeVirt on the cluster
* run KubeVirt functional tests against the cluster

TODOs:
* build latest KubeVirt from scratch (ATM I'm using a version pushed to my own docker repository)
* create a new gcp-nested-virt template based on the cluster-launch-installer-custom-test-image.yaml template
* move the test shell code in the config back to the kubevirt/hack/ci/test.sh file

### CODE REVIEW

File: ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- as: "kubevirt-e2e-test"
+  commands: |
+    set -e
+    set -o pipefail
+
+    export GIMME_GO_VERSION=1.12.8
+    source /etc/profile.d/gimme.sh
+
+    export KUBEVIRT_PROVIDER=external
+    ./hack/generate.sh
```
Comment (qinqon) [inline ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml]: Why don't you call make here ?

Comment (dhiller) [reply ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml]: Because bazel is not available.

Comment (dhiller) [reply ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml]: BTW: now bazel is available as we can use the kubevirt-builder, but still docker is not available, so we can't use anything that uses `hack/dockerized`

File: ci-operator/jobs/kubevirt/kubevirt/kubevirt-kubevirt-master-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        - name: CLUSTER_TYPE
+          value: gcp
+        - name: CONFIG_SPEC
+          valueFrom:
+            configMapKeyRef:
+              key: kubevirt-kubevirt-master.yaml
+              name: ci-operator-master-configs
+        - name: JOB_NAME_SAFE
+          value: kubevirt-e2e-test
+        - name: TEST_COMMAND
```
Comment (droslean) [inline ci-operator/jobs/kubevirt/kubevirt/kubevirt-kubevirt-master-presubmits.yaml]: All of those should be inside the template and you should add here only your test commands. 

Also, you need to do some changes in `prowgen` as well to include the new template. see [here](https://github.com/openshift/ci-tools/blob/master/cmd/ci-operator-prowgen/main.go#L300-L347) and [here](https://github.com/openshift/ci-tools/blob/master/pkg/api/config.go#L262-L329)

File: ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tag_specification:
+  name: "4.2"
+  namespace: ocp
+tests:
```
Comment (droslean) [inline ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml]: after you do the `prowgen` changes, you will be able to configure your job here. Then prowgen will be able to create it

**Comment by dhiller** [comment]
For reference to the discussion: this PR holds commits from [this other PR](https://github.com/dhiller/release/pull/1)

File: ci-operator/jobs/kubevirt/kubevirt/kubevirt-kubevirt-master-presubmits.yaml (hunk 2 above)
Comment (dhiller) [reply ci-operator/jobs/kubevirt/kubevirt/kubevirt-kubevirt-master-presubmits.yaml]: We've created a new template based on `cluster-launch-installer-libvirt-e2e.yaml`, I dont see any reference for that one? Does that mean that for that template the changes are pending or aren't they required as this is too special?

**Comment by dhiller** [comment]
So, as stated in the forum-testplatform chat I got problems in debugging this one, could someone help me out with what is required and how to do this?

**Comment by dhiller** [comment]
Should work now as #6594 is merged.

/retest

**Comment by dhiller** [comment]
@sallyom if you have time, you might assist me in debugging this error. The template is running so far, although the setup container seems to run onto an error when trying to propagate the ssh credentials?

```
gcloud compute scp --quiet --project openshift-gce-devel-ci --zone us-east1-c --recurse /home/packer/kubevirt 'packer@ci-op-hfbwjinb-a2561:~/'
```

```
Waiting for SSH key to propagate.
ssh: connect to host 104.196.42.175 port 22: Connection timed out
ERROR: (gcloud.compute.scp) Could not SSH into the instance.  It is possible that your SSH key has not propagated to the instance yet. Try running this command again.  If you still cannot connect, verify that the firewall and instance are set to accept ssh traffic.
```

Complete stack trace here:  https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/6146/rehearse-6146-pull-ci-kubevirt-kubevirt-master-kubevirt-e2e-test/34

IDK, maybe we should loop the scp and retry three times or so? Any ideas are welcome. BTW I may be out soon, but we may setup a talk tomorrow if you like?

**Comment by sallyom** [comment]
@dhiller i'll take a look, catch you tomorrow : )

**Comment by sallyom** [comment]
this will require something like this: https://github.com/dhiller/release/pull/2  and this: https://github.com/kubevirt/kubevirt/pull/2991

**Comment by fabiand** [comment]
@dhiller thoughts?

**Comment by dhiller** [comment]
@fabiand Yes,many ;) Seriously, we are currently discussing in the mentioned PRs.

File: ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml (hunk 1 above)
Comment (rmohr) [reply ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml]: We can directly use the builder image as a base, so the good thing is that we don't need to extra start the builder container to use its content. :+1:

**Comment by dhiller** [comment]
Waiting for kubevirt/kubevirt#3064 to get merged, so that we can have the image build complete.

**Comment by dhiller** [comment]
@droslean can you please re-review? Thanks!

**Pull request merged**